src/Security/Authentication/OpenIdConnect/src/OpenIdConnectConfigureOptions.cs

```diff
@@ -0,0 +1,170 @@
+        options.ForwardForbid = configSection[nameof(options.ForwardForbid)];
+        options.ForwardSignIn = configSection[nameof(options.ForwardSignIn)];
+        options.ForwardSignOut = configSection[nameof(options.ForwardSignOut)];
+        options.GetClaimsFromUserInfoEndpoint = bool.TryParse(configSection[nameof(options.GetClaimsFromUserInfoEndpoint)], out var getClaimsFromUserInfoEndpoint)
+            ? getClaimsFromUserInfoEndpoint
+            : options.GetClaimsFromUserInfoEndpoint;
+        options.MapInboundClaims = bool.TryParse(configSection[nameof(options.MapInboundClaims)], out var mapInboundClaims)
+            ? mapInboundClaims
+            : options.MapInboundClaims;
+        options.MaxAge = TimeSpan.TryParse(configSection[nameof(options.MaxAge)], out var maxAge)
+            ? maxAge
+            : options.MaxAge;
+        options.MetadataAddress = configSection[nameof(options.MetadataAddress)];
+        options.NonceCookie = GetCookieFromConfig(configSection.GetSection(nameof(options.NonceCookie)), options.NonceCookie);
+        options.Prompt = configSection[nameof(options.Prompt)];
+        options.RefreshInterval = TimeSpan.TryParse(configSection[nameof(options.RefreshInterval)], out var refreshInterval)
+            ? refreshInterval
+            : options.RefreshInterval;
+        options.RefreshOnIssuerKeyNotFound = bool.TryParse(configSection[nameof(options.RefreshOnIssuerKeyNotFound)], out var refreshOnIssuerKeyNotFound)
+            ? refreshOnIssuerKeyNotFound
+            : options.RefreshOnIssuerKeyNotFound;
+        options.RemoteAuthenticationTimeout = TimeSpan.TryParse(configSection[nameof(options.RemoteAuthenticationTimeout)], out var remoteAuthenticationTimeout)
+            ? remoteAuthenticationTimeout
+            : options.RemoteAuthenticationTimeout;
+        options.RemoteSignOutPath = new PathString(configSection[nameof(options.RemoteSignOutPath)] ?? options.RemoteSignOutPath);
+        options.RequireHttpsMetadata = bool.TryParse(configSection[nameof(options.RequireHttpsMetadata)], out var requireHttpsMetadata)
+            ? requireHttpsMetadata
+            : options.RequireHttpsMetadata;
+        options.Resource = configSection[nameof(options.Resource)];
+        options.ResponseMode = configSection[nameof(options.ResponseMode)] ?? options.ResponseMode;
+        options.ResponseType = configSection[nameof(options.ResponseType)] ?? options.ResponseType;
+        options.ReturnUrlParameter = configSection[nameof(options.ReturnUrlParameter)] ?? options.ReturnUrlParameter;
+        options.SaveTokens = bool.TryParse(configSection[nameof(options.SaveTokens)], out var saveTokens)
+            ? saveTokens
+            : options.SaveTokens;
+        var scopes = configSection.GetSection(nameof(options.Scope)).GetChildren().Select(scope => scope.Value).OfType<string>();
+        foreach (var scope in scopes)
+        {
+            options.Scope.Add(scope);
```

Comment:
I'm not sure if it's needed, but there's no way to remove the default scopes right now.
https://github.com/dotnet/aspnetcore/blob/5e86e1b92f7c907dd20767ce7c4546de0740263a/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectOptions.cs#L43-L44

Reply:
The `openid` scope is required in OIDC and I think scenarios where the user would not want a `profile` scope too are rare to justify supporting modification here.

Reply:
should Scope be cleared before adding new values? (if there are any)